test/declaration_test.ts

```diff
@@ -255,6 +255,39 @@ class X {
   // @Ignore
   X.fakeConstructor$();
   external x();
+}`);
+    });
+    it('should emit extension methods to return Futures from methods that return Promises', () => {
+      expectTranslate(`declare interface MyMath {
+        randomInRange(start: number, end: number): Promise<number>;
+      }`).to.equal(`import "dart:async" show Completer;
+
+@anonymous
+@JS()
+abstract class MyMath {
+  external dynamic /*Promise<num>*/ randomInRange(num start, num end);
```

Comment:
call this method
JS$randomInRange 
and the JS interop code should still do the right thing.
@natebosch
The 
`JS$`
ugly prefix works in both dart2js and ddc, right?
It is a workaround to let you handle most cases where allowed JS identifiers and allowed JS identifiers don't align.
In this case it could work well to give the more important Future based Dart API a nice API and the Ugly JS promise based API ugly names.

Reply:
I've never seen `JS$`...

Reply:
It looks like it is still referenced by tests/lib_2/html/js_util_test.dart
assuming those tests are still executing it should work.
if it doesn't work, the other equally reasonable option is to call
js_util.callMethod directly from the extension methods and omit the API with promises completely from the generated code.
Put another way, with extension methods you don't need the external method from @JS classes at all.
Instead you could create empty
```dart
@JS
class MyClass {
}
@JS
class MyClass2 {
}
```
classes and provide all the implementations in extension methods. In some ways this would be cleaner as all the API definition would be static instead of mixing the psuedo-dynamic definitions from @JS class methods with static methods from the extensions.
This would also allow you to define generic methods and convert List<Object> to List<yourFavoriteType> objects on the API boundary which are both things you can't do with the external methods.

Reply:
Fyi @jakemac53 this is a plausible strawman alternative to `JsAny`.